airbyte-integrations/connectors/source-trustpilot/source_trustpilot/source.py

```diff
@@ -0,0 +1,178 @@
+    3. `stream_slices` output is pretty same to `get_stream_state` output.
+    """
+
+    start_datetime: Union[MinMaxDatetime, str]
+    cursor_field: Union[InterpolatedString, str]
+    datetime_format: str
+    config: Config
+
+    _start_datetime_value: datetime.datetime = field(repr=False, default=None)
+    _stream_slice_value: datetime.datetime = field(repr=False, default=None)
+    _first_record_value: datetime.datetime = field(repr=False, default=None)
+
+    _cursor_field_name: str = field(repr=False, default=None)
+    _new_cursor_field_name: str = field(repr=False, default=None)
+    _old_cursor_field_name: str = field(repr=False, default=None)
+
+    def __post_init__(self, options: Mapping[str, Any]):
+        if not isinstance(self.start_datetime, MinMaxDatetime):
+            self.start_datetime = MinMaxDatetime(self.start_datetime, options)
+
+        self._timezone = datetime.timezone.utc
+        self._parser = DatetimeParser()
+
+        if not self.start_datetime.datetime_format:
+            self.start_datetime.datetime_format = self.datetime_format
+
+        self._start_datetime_value = self.start_datetime.get_datetime(self.config)
+
+        self.cursor_field = InterpolatedString.create(self.cursor_field, options=options)
+
+        self._cursor_field_name = self.cursor_field.eval(self.config)
+        self._new_cursor_field_name = 'new_' + self._cursor_field_name
+        self._old_cursor_field_name = 'old_' + self._cursor_field_name
+
+    def _format_datetime(self, dt: datetime.datetime):
+        return self._parser.format(dt, self.datetime_format)
+
+    def parse_date(self, date: str) -> datetime.datetime:
+        return self._parser.parse(date, self.datetime_format, self._timezone)
+
+    def update_cursor(self, stream_slice: StreamSlice, last_record: Optional[Record] = None):
+        if last_record is not None:
+            if self._first_record_value is None:
+                tmp = last_record.get(self.cursor_field.eval(self.config))
+                self._first_record_value = self.parse_date(tmp)
+        elif stream_slice is not None:
+            tmp = stream_slice.get(self._new_cursor_field_name)
+            self._stream_slice_value = self.parse_date(tmp)
+
+    def get_stream_state(self) -> StreamState:
+        return {
+            self._old_cursor_field_name: self._format_datetime(
+                self._stream_slice_value if self._stream_slice_value is not None else self._start_datetime_value),
+
+            self._new_cursor_field_name: self._format_datetime(self._first_record_value if self._first_record_value is not None else (
+                self._stream_slice_value if self._stream_slice_value is not None else self._start_datetime_value
+            ))
+        }
+
+    def stream_slices(self, sync_mode: SyncMode, stream_state: Mapping[str, Any]) -> Iterable[StreamSlice]:
+        return [self.get_stream_state()]
```

Comment:
Same here unit tests for functions and class.

Reply:
Okay, great suggestion!
Thank You for advice!

Reply:
Did it.